### ISSUE TITLE
Block border is hard to see and can cause confusion interpreting the code

### ISSUE THREAD

**Issue by microbit-carlos** [microsoft/pxt-microbit#4962]
This code snippet was posted on Reddit a few days ago and it illustrates how hard it is to spot a code bug of this type in the blocks view.
This is less of a problem when the input blocks are a different colour, but when the colour is the same it's really hard to see the block border and tell them apart.

https://www.reddit.com/r/ProgrammerHumor/comments/118xona/i_dont_know_how_to_use_python_or_javascript_at/

![kgj3qkns3sja1](https://user-images.githubusercontent.com/29712657/221584729-98195026-2e63-4288-aeab-2bd598de3b63.jpg)

I would argue that in that picture of a laptop monitor the block border is more noticeable than it is in my own laptop screen. Without knowing what to look for these two blocks look almost indistinguishable to me:
https://makecode.microbit.org/_0cF9euXoRF8R
<img width="511" alt="image" src="https://user-images.githubusercontent.com/29712657/221585498-712b3ce8-d506-4bde-92bd-57b01ff0f977.png">

Can we do something to increase the contrast between blocks of the same colour?

**Comment by abchatra** [comment]
No plans to address.

**Comment by microbit-carlos** [comment]
We'd like this to be reconsidered for the next release if possible.

**Comment by abchatra** [comment]
@riknoll anything could be done here cheaply?

**Comment by riknoll** [comment]
I've been trying a few experiments this morning

For reference, this is the current way this is rendered:

![image](https://github.com/user-attachments/assets/367422f7-c3d9-43fd-adf3-f2df8b082e92)

Here is the same rendering but with all of the block borders slightly darker:

![image](https://github.com/user-attachments/assets/4d4efd07-3a61-484c-bb3d-1d4c582ca827)

Here is the default border color but with slightly more padding added to each block:

![image](https://github.com/user-attachments/assets/a64b3129-539b-4963-8830-b70f7fffe327)

...and here is the increased padding with the darker border:

![image](https://github.com/user-attachments/assets/0cb6dcb7-619b-4728-b995-6409342bd198)


Would love to hear thoughts @abchatra @Jaqster @microbit-carlos

**Comment by Jaqster** [comment]
I like the darker border option.  I worry about increased padding that makes the blocks fatter and will take up more screen real estate.

**Comment by riknoll** [comment]
my personal favorite is same border, increased padding. i think the darker border with no increased padding is very hard to read

**Comment by Jaqster** [comment]
instead of a dark border, what if we did a lighter border?

**Comment by thsparks** [comment]
Padding seems very subtle to me, not sure I'd have even noticed it without the label. Could you show the side-by-side comparison from the original bug with the extra padding (the two expressions that are similar)? I'd be curious to see how they look visually next to each other.

I think the darker border looks nice though.

**Comment by martinwork** [comment]
In live, hovering highlights the parts, 
![image](https://github.com/user-attachments/assets/d705c355-a9b4-4d64-80a0-651301f989b5)
![image](https://github.com/user-attachments/assets/c8628400-7488-47bb-acc4-139f968a63ad)

but in beta, only the dropdowns are highlighted.
![image](https://github.com/user-attachments/assets/d113122c-0e4b-48fd-b076-d94da32e2e3a)

**Comment by Jaqster** [comment]
> instead of a dark border, what if we did a lighter border?

In Martin's screenshot, I do like the light border - makes it very clear.  What if we did this only for same color nested blocks?  Would it look too busy if we had multiple nests?  
![image](https://github.com/user-attachments/assets/80096549-6e40-4086-8bec-a44934495290)

**Comment by jaustin** [comment]
> In Martin's screenshot, I do like the light border - makes it very clear. What if we did this only for same color nested blocks? Would it look too busy if we had multiple nests?

We discussed this internally and @microbit-carlos may have a lingering screenshot from playing with the Inspector to test it. We Iiked the lighter colour

I think it would be great if "light border for same colour nested blocks" was do-able. It feels like a far clearer way to do it.

If we didn't have the ability to only turn lighter borders on for same-coloured-blocks, I think lighter, not darker, might be a good tweak to apply to the borders on the purple blocks?

**Comment by riknoll** [comment]
i don't think that a lighter border will work for every color of block, but i agree it looks better for the relatively-dark math category. 

we already do some color altering based on relative luminance for shadow blocks, i'll see if there is some way for me to apply that to all nested blocks. will probably have to modify the renderer a bit

**Comment by riknoll** [comment]
here are some options for lightness of the borders

option 1 (darkest):
![image](https://github.com/user-attachments/assets/c16af10b-1455-485c-83f7-eb5994ff8a63)

option 2:
![image](https://github.com/user-attachments/assets/57dc689b-f42d-4e27-807d-bb3691baad7b)

option 3 (lightest):
![image](https://github.com/user-attachments/assets/443561f3-3e22-4b6e-8449-7636e52e2ba8)

**Comment by riknoll** [comment]
Same options, but with 1 extra pixel of padding:

option 1 (darkest):

![image](https://github.com/user-attachments/assets/33b64472-b3fe-4f84-bee8-8e80f70abb1a)

option 2:

![image](https://github.com/user-attachments/assets/110b1c14-40b9-45f0-b285-1155f5913611)

option 3 (lightest):

![image](https://github.com/user-attachments/assets/f0289fea-8e40-4f7b-a5f4-203015f30d8c)

**Comment by riknoll** [comment]
same options, but with the extra padding and very slightly thicker borders on all blocks

option 1 (darkest):

![image](https://github.com/user-attachments/assets/476adde1-a869-4277-9874-16e46f66ca04)

option 2:

![image](https://github.com/user-attachments/assets/90790a9b-abab-40d7-8348-5c67acbdffdc)

option 3 (lightest):

![image](https://github.com/user-attachments/assets/f82be50f-8c4f-46ac-a2af-290fdfff047e)

**Comment by Jaqster** [comment]
My personal preference would be Option 1 or 2 for the color (I think even option 1, the darkest shows up well).  I don't think we need extra padding or to change the line thickness.

**Comment by thsparks** [comment]
My vote would be for 2, runner-up choice would be number 3

**Comment by microbit-carlos** [comment]
Option 2 or 3 would be my vote. With the lighter border I don't think we need the extra padding either.
The padding was useful when there was less contrast between the border and the block colours.


> we already do some color altering based on relative luminance for shadow blocks, i'll see if there is some way for me to apply that to all nested blocks. will probably have to modify the renderer a bit

If this was possible and it could automatically render the border colour (light or dark) with the best contrast based on the block colour, that would be fantastic!

**Comment by jaustin** [comment]
We've discussed here and would be pleased to see this merged. If we have to come down on a preference, ours is 2, too!

**Comment by riknoll** [comment]
I did indeed go with option 2 in the pr!
